dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/MongoDBRyaDAO.java

```diff
@@ -164,26 +174,31 @@ public void add(final Iterator<RyaStatement> statement) throws RyaDAOException {
         final List<DBObject> dbInserts = new ArrayList<DBObject>();
         while (statement.hasNext()){
             final RyaStatement ryaStatement = statement.next();
-            final DBObject insert = storageStrategy.serialize(ryaStatement);
-            dbInserts.add(insert);
-
-            try {
-                for (final RyaSecondaryIndexer index : secondaryIndexers) {
-                    index.storeStatement(ryaStatement);
+            final boolean canAdd = DocumentVisibilityUtil.doesUserHaveDocumentAccess(auths, ryaStatement.getColumnVisibility());
+            if (canAdd) {
+                final DBObject insert = storageStrategy.serialize(ryaStatement);
+                dbInserts.add(insert);
+
+                try {
+                    for (final RyaSecondaryIndexer index : secondaryIndexers) {
+                        index.storeStatement(ryaStatement);
+                    }
+                } catch (final IOException e) {
+                    log.error("Failed to add: " + ryaStatement.toString() + " to the indexer");
                 }
-            } catch (final IOException e) {
-                log.error("Failed to add: " + ryaStatement.toString() + " to the indexer");
             }
-
         }
         coll.insert(dbInserts, new InsertOptions().continueOnError(true));
     }
 
     @Override
     public void delete(final RyaStatement statement, final MongoDBRdfConfiguration conf)
             throws RyaDAOException {
-        final DBObject obj = storageStrategy.getQuery(statement);
-        coll.remove(obj);
+        final boolean canDelete = DocumentVisibilityUtil.doesUserHaveDocumentAccess(auths, statement.getColumnVisibility());
+        if (canDelete) {
+            final DBObject obj = storageStrategy.getQuery(statement);
+            coll.remove(obj);
```

Comment:
This PR was rebased off master before posting so I don't think I took out anybody's code.  I'll rebase again if I make any more updates.